### ISSUE TITLE
Can't exit modal in 0.9.47

### ISSUE THREAD

**Issue by magnus-staberg** [Hammerspoon/hammerspoon#979]
The code below works in 0.9.46 but not 0.9.47:
```
modal = hs.hotkey.modal.new({'ctrl'}, 'M')
modal:bind({}, 'escape', function() modal:exit() end)
```
It gives
`hotkey: The hotkey was deleted, cannot enable it`
in console. The invocation of `modal:exit` is the culprit.

Am I missing something or is it a bug? 
Couldn't find anything in the release notes.

**Comment by cro** [comment]
I'm seeing this behavior also.

**Comment by asmagill** [comment]
My experience with this is that while the error message appears, it does clear the modal keys and execute `modal:exited()` (if it exists) properly... does this match your experience or are you seeing something not being executed properly or keys not unset?  I ask because it may affect what I should be looking for...

**Comment by magnus-staberg** [comment]
I don't know if it executes `modal:exited()` but it doesn't exit properly.

**Comment by asmagill** [comment]
Are you able to build a custom build of Hammerspoon and test the fix I posted in Pull #988?

Actually, it might be easier to just apply the change yourself, since it's just one line to the `hs.hotkey` init.lua file...  Try the following:

Locate the Hammerspoon application icon in your Finder... it's probably in "Applications" unless you've moved it somewhere else.  Right click on the icon and select "Show Package Contents".  Double click on the "Contents" folder, the "Resources" folder, the "extensions" folder, the "hs" folder, then the "hotkey" folder (they should show up in that order).  Select the "init.lua" file and open it up in an editor... double clicking on it might open one, or you can right click on it and select "Open With" and choose an editor... TextEdit works fine for this if you don't have anything else.

Look for the following code at around line 51... 
~~~lua
local function enable(self,force,isModal)
  -- force is there just in case it'll be needed down the line; not exposed as public API for now
  -- isModal is only used to differentiate loglevel (modals can cause a lot of shadowing/unshadowing)

  -- this ensures "nested shadowing" behaviour; i.e. can't re-enable a hotkey that is currently shadowed (unless force),
  -- must wait for the current top dog to get disabled
  if not force and self.enabled then log.v('Hotkey already enabled') return self end
  local idx = self.idx
  if not idx or not hotkeys[idx] then log.e('The hotkey was deleted, cannot enable it') return end
  for i=#hotkeys[idx],1,-1 do
    local hk=hotkeys[idx][i]
    if hk==self then tremove(hotkeys[idx],i) -- this hotkey will go to the top of the stack
    elseif hk.enabled then log.i('Disabled previous hotkey '..hk.msg) end --shadow previous hotkeys
    hk._hk:disable() --objc
  end
  self.enabled = true
  local returnVal = self._hk:enable() --objc
  log[isModal and 'df' or 'f']('Enabled hotkey %s%s',self.msg,isModal and ' (in modal)' or '')
  tinsert(hotkeys[idx],self) -- bring to the top of the stack
  return returnVal
end
~~~

It's actually the second to last line we care about, and it's at line 73, I believe... change `return returnVal` to `return self`

Save the file and completely quit and restart Hammerspoon.

This tweak has caused the error messages to disappear completely from my console and everything in my setup seems to be working fine with it, though I hadn't been experiencing anything consistent enough to be a problem with modal hotkey's before (mostly just the annoyance of the message), so I'd like to hear how it affects your problem as well.

**Comment by magnus-staberg** [comment]
Yep, it's now working. I edited the return value as you said. Thanks a lot for your detailed response and fix 👍 . I guess this issue should be closed when the pull request is merged.

**Comment by cmsj** [comment]
This should be resolved now that #988 has merged. 0.9.48 will be released soon with the fix. Sorry everyone!
